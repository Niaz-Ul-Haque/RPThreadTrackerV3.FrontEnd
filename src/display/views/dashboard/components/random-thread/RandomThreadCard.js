import React from 'react';
import PropTypes from 'prop-types';
import { CardHeader, CardBlock } from 'reactstrap';
import Card from '../../../../shared/styled/Card';
import RandomThreadDisplay from './RandomThreadDisplay';

const propTypes = {
	randomThread: PropTypes.shape({}).isRequired,
	generateRandomThread: PropTypes.func.isRequired
};

const RandomThreadCard = (props) => {
	const { generateRandomThread, randomThread } = props;
	return (
		<Card className="random-thread-generator-card" data-spec="random-thread-generator-card">
			<CardHeader data-spec="random-thread-generator-header">
				<i
					className="fas fa-random"
					data-spec="random-thread-generator-icon"
				/> Random Thread Generator
			</CardHeader>
			<CardBlock className="card-body">
				<button
					type="button"
					className="btn btn-primary"
					onClick={generateRandomThread}
					data-spec="random-thread-generator-button"
				>
					Generate
				</button>
				<RandomThreadDisplay data-spec="random-thread-display" threadData={randomThread} />
			</CardBlock>
		</Card>
	);
};
RandomThreadCard.propTypes = propTypes;
export default RandomThreadCard;
