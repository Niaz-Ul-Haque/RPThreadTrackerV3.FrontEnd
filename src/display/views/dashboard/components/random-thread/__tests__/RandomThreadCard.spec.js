// #region imports
import React from 'react';
import { shallow } from 'enzyme';
import { getSpecWrapper } from '~/testhelpers/helpers.unit';
import RandomThreadCard from '../RandomThreadCard';
// #endregion imports

// #region mocks
jest.mock('../RandomThreadDisplay', () => 'RandomThreadDisplay');
// #endregion mocks

const createTestProps = (propOverrides) => ({
	// common props
	generateRandomThread: jest.fn(),
	randomThread: {
		thread: {
			userTitle: 'Test Title'
		},
		status: {}
	},
	...propOverrides
});

describe('rendering', () => {
	describe('snapshots', () => {
		it('should render valid snapshot', () => {
			const props = createTestProps();
			const jsx = <RandomThreadCard {...props} />;
			const element = shallow(jsx);
			expect(element).toMatchSnapshot();
		});
	});
});

describe('behavior', () => {
	describe('generateRandomThread', () => {
		it('should be triggered on button click', () => {
			const generateRandomThread = jest.fn();
			const props = createTestProps({
				generateRandomThread
			});
			const jsx = <RandomThreadCard {...props} />;
			const element = shallow(jsx);
			const button = getSpecWrapper(element, 'random-thread-generator-button');
			button.simulate('click');
			expect(generateRandomThread).toHaveBeenCalledTimes(1);
		});
	});
});
