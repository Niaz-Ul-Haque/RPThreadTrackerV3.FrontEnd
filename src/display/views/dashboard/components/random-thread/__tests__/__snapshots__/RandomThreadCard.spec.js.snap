// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering snapshots should render valid snapshot 1`] = `
<Styled(Card)
  className="random-thread-generator-card"
  data-spec="random-thread-generator-card"
>
  <CardHeader
    data-spec="random-thread-generator-header"
    tag="div"
  >
    <i
      className="fas fa-random"
      data-spec="random-thread-generator-icon"
    />
     Random Thread Generator
  </CardHeader>
  <CardBlock
    className="card-body"
    tag="div"
  >
    <button
      className="btn btn-primary"
      data-spec="random-thread-generator-button"
      onClick={[MockFunction]}
      type="button"
    >
      Generate
    </button>
    <RandomThreadDisplay
      data-spec="random-thread-display"
      threadData={
        Object {
          "status": Object {},
          "thread": Object {
            "userTitle": "Test Title",
          },
        }
      }
    />
  </CardBlock>
</Styled(Card)>
`;
